src/number_format_base.tsx

```diff
@@ -217,7 +217,9 @@ export default function NumberFormatBase<BaseType = InputAttributes>(
 
   useIsomorphicLayoutEffect(() => {
     const input = focusedElm.current;
-    if (formattedValue !== lastUpdatedValue.current.formattedValue && input) {
+    const activeElement = document.activeElement;
+
+    if (input === activeElement && formattedValue !== lastUpdatedValue.current.formattedValue && input) {
```

Comment:
Humm, @kasperdun can you share an example where this is breaking? Basically focusedElm.current is set only when that input is focused. As we set the focusedElm.current to undefined when a blur is called.